apps/annotations/models.py

```diff
@@ -98,20 +98,29 @@ def add_tag(self, tag: Tag, team: Team, added_by: CustomUser):
         self.tags.add(tag, through_defaults={"team": team, "user": added_by})
 
     def get_linked_tags(self):
-        return self.tags.all()
+        tags = []
+        for tag in self.tags.all():
+            bage_color = "badge-neutral"
+            if tag.is_system_tag:
+                bage_color = "badge-error" if tag.category == TagCategories.SAFETY_LAYER_RESPONSE else "badge-warning"
```

Comment:
It can be dangerous to use css classes in Python since the post processor will remove classes that aren't used and it doesn't currently check python files for usages.

What about doing this logic in the template and just using 'tags_json' instead of this method? I think we could get rid of this method entirely if we did that.

Reply:
https://github.com/dimagi/open-chat-studio/pull/707/commits/41776c6d115b3d487621f811b232d70b966fb349